### PR TITLE
Update: Auto insert page-link on navigation sidebar.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR auto inserts page links on the navigation sidebar.
cc: @jasmussen, @jameskoster

### CODE REVIEW

**Comment by jameskoster** [comment]
Nice, this makes it _much_ easier to add Page Links to the menu which is a very common use case.

**Before**

https://user-images.githubusercontent.com/846565/222809678-1d4469a6-ca89-46a8-ab45-a8099c94d86e.mp4



**After**

https://user-images.githubusercontent.com/846565/222809744-2edc8624-4310-45c4-b70e-da560a941a44.mp4

**Comment by Mamaduka** [comment]
@jorgefilipecosta, @jameskoster is this alternative to #48724?

File: packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js [hunk 1]
```diff
@@ -41,6 +45,29 @@ function SidebarNavigationScreenWrapper( { children, actions } ) {
 	);
 }
 
+function PageLinkInserter( { rootClientId } ) {
+	const { insertBlock } = useDispatch( blockEditorStore );
+	const { getBlockOrder } = useSelect( blockEditorStore );
+	const insertPageLink = useCallback( () => {
```
Comment (Mamaduka) [inline packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/index.js:51]: Nit: Probably no need for `useCallback` here. The `SidebarButton` component isn't memoized and will re-render whenever its parent re-renders. So we don't need to pass a referentially stable callback to it.

**Comment by jasmussen** [comment]
> is this alternative to https://github.com/WordPress/gutenberg/pull/48724?

No, this is in addition to. 48724 is important because when building any navigation menu, you're likely to be using Page Links. 

The way it behaves is that if you only ever have menu items in your menu, you get this "light weight" navigation building flow where navigation items are inserted directly, skipping the quick inserter. This is similar to how the Buttons block defaults to inserting buttons, skipping the need to pick "button" from the inserter.

As soon as you have other block types inside the navigation wrapper, such as a site logo or a search field, you get the regular quick inserter, which per 48724 should still pin "page links" for quick access.

**Comment by jasmussen** [comment]
Speaking of: this PR is working well for me in that it defaults to the Page Links:

Screenshot 2023-03-06 at 09 48 08

It doesn't have the same behavior as the navigation block, though, [as described here](https://github.com/WordPress/gutenberg/pull/48745#issuecomment-1455717972), where as soon as you have a block inside the nav wrapper that isn't a menu item, it falls back to the regular inserter. 

It should probably be the same behavior here. Can we use the exact same componentry as the off-canvas navigation editor has? Ideally the code would be reused.

**Comment by Mamaduka** [comment]
> As soon as you have other block types inside the navigation wrapper, such as a site logo or a search field, you get the regular quick inserter, which per 48724 should still pin "page links" for quick access.

I don't see the logic for this behavior in the code. Currently, PR replaces block inserter with a custom inserter that only handles Page Links.

**Comment by jorgefilipecosta** [comment]
> I don't see the logic for this behavior in the code. Currently, PR replaces block inserter with a custom inserter that only handles Page Links.

We currently don't have this logic, but it is a simple change to include. I'm unsure if this PR is needed, given that https://github.com/WordPress/gutenberg/pull/48740 was already merged.

**Comment by Mamaduka** [comment]
I've removed backport label until we have a final decision for this enhancement.

**Comment by paaljoachim** [comment]
Using WP 6.2 beta 4
Beta tester plugin.
Twenty Twenty Three
Brave browser

Before:
Going to the site editor.
Selecting Navigation.
Clicking the + sign (appender) I see this:
Screenshot 2023-03-07 at 10 43 14

After:
Using the PR Gutenberg build. Clicking the + sign (appender) I see a Page Link is automatically added to the menu. I see that I can easily choose between various pages (posts do not show up - even searching for a post it will not show up).

Screenshot 2023-03-07 at 10 47 23

I am puzzled that clicking the appender a Page Link is automatically added. I would believe that we would through consistency use a similar method as in template page Navigation would use. Making the Browse sidebar Navigation a kind of duplicate of the in page template Navigation.

Example from the in page template Navigation.
Clicking the appender an "Add link" is added and I see the following.

Screenshot 2023-03-07 at 10 54 36

Btw
In the Search field I can also search for a post and it will show up. Search needs to work for pages and posts.

**Comment by jameskoster** [comment]
> It doesn't have the same behavior as the navigation block, though, https://github.com/WordPress/gutenberg/pull/48745#issuecomment-1455717972, where as soon as you have a block inside the nav wrapper that isn't a menu item, it falls back to the regular inserter.

I don't really understand why we do this 🤔 Adding a Page Link is inherently the most common behavior, regardless of whether the menu contains other block types, no?

**Comment by ntsekouras** [comment]
With this PR, we can only add the `page-link` and nothing else.. Is this what we want? Especially now that [the page link is prioritized](https://github.com/WordPress/gutenberg/pull/48752) in the inserter initial results?

**Comment by jorgefilipecosta** [comment]
> With this PR, we can only add the `page-link` and nothing else.. Is this what we want? Especially now that [the page link is prioritized](https://github.com/WordPress/gutenberg/pull/48752) in the inserter initial results?

I think the idea, yes, the idea was only to insert page links. If people want other blocks, the page link creation popover allows changing to other blocks.

**Comment by jorgefilipecosta** [comment]
Closing this PR, as right now the inserter is disable and what the inserter will look like is not yet decided.

**Pull request closed without merging**